### ISSUE TITLE
DeeplabV3+ Training on custom dataset

### ISSUE THREAD

**Issue by parth29-vc** [keras-team/keras#15916]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- Platform and Distribution: Colab
- TensorFlow/Keras version (use command below): 2.7.0
- Python version: 3.8
- CUDA/cuDNN version: 11.2
- GPU model and memory: Tesla P100, 16 GB memory

**Current situation**
 Worked on Semantics segmentation using TensorFlow's DeepLabv3+ with the Resnet versions backbone architecture. We've trained it for our custom dataset which includes satellite images of Golf Courses extracted from the Google Map API. Our Dataset includes 109 samples for training and we have trained it and got a decent amount of accuracy (~93%) but the results are not as expected.

**Current behavior**
Currently, we've referenced the test set which includes similar types of images but the results are not as good as expected. For the sake of curiosity, we have checked by inferencing on Train images also but that also not promising.

**Expected behavior**
Expected results with this accuracy should be descent as trained on the other custom dataset.

The gist for the code snippet is available here. [Gist](https://gist.github.com/parth29-vc/29029bd2989bfc994fedfb976af65b16)

**Comment by sushreebarsa** [comment]
@parth29-vc 
In order to reproduce the issue reported here, could you please provide the standalone code and the dataset ? Thanks!
